src/editor/editor.cpp

```diff
@@ -0,0 +1,375 @@
+//  SuperTux
+//  Copyright (C) 2015 Hume2 <teratux.mail@gmail.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "editor/editor.hpp"
+
+#include "audio/sound_manager.hpp"
+#include "control/input_manager.hpp"
+#include "editor/layer_icon.hpp"
+#include "gui/mousecursor.hpp"
+#include "gui/menu_manager.hpp"
+#include "object/camera.hpp"
+#include "object/player.hpp"
+#include "object/tilemap.hpp"
+#include "supertux/menu/menu_storage.hpp"
+#include "supertux/menu/editor_menu.hpp"
+#include "supertux/menu/editor_levelset_select_menu.hpp"
+#include "supertux/game_manager.hpp"
+#include "supertux/game_object.hpp"
+#include "supertux/level.hpp"
+#include "supertux/level_parser.hpp"
+#include "supertux/levelset_screen.hpp"
+#include "supertux/moving_object.hpp"
+#include "supertux/savegame.hpp"
+#include "supertux/fadein.hpp"
+#include "supertux/screen.hpp"
+#include "supertux/screen_fade.hpp"
+#include "supertux/screen_manager.hpp"
+#include "supertux/sector.hpp"
+#include "supertux/tile.hpp"
+#include "supertux/tile_manager.hpp"
+#include "supertux/world.hpp"
+#include "video/surface.hpp"
+
+Editor::Editor() :
+  level(),
+  world(),
+  levelfile(),
+  quit_request(false),
+  newlevel_request(false),
+  reload_request(false),
+  reactivate_request(false),
+  deactivate_request(false),
+  save_request(false),
+  test_request(false),
+  m_savegame(),
+  currentsector(),
+  levelloaded(false),
+  leveltested(false),
+  tileset(NULL),
+  inputcenter(),
+  tileselect(),
+  layerselect(),
+  enabled(false),
+  bgr_surface()
+{
+  bgr_surface = Surface::create("images/background/forest1.jpg");
+}
+
+Editor::~Editor()
+{
+
+}
+
+void Editor::draw(DrawingContext& context)
+{
+  if (levelloaded) {
+    currentsector->draw(context);
+  } else {
+    context.draw_surface_part(bgr_surface, Rectf(Vector(0, 0), bgr_surface->get_size()),
+                              Rectf(Vector(0, 0), Vector(SCREEN_WIDTH, SCREEN_HEIGHT)), -100);
+  }
+  inputcenter.draw(context);
+  tileselect.draw(context);
+  layerselect.draw(context);
+  MouseCursor::current()->draw(context);
+}
+
+void Editor::update(float elapsed_time)
+{
+  // Reactivate the editor after level test
+  if (leveltested) {
```

Comment:
I think all the following if's could use a switch-case instead. 

You could have a current_request variable of an enum.